src/status_im2/contexts/profile/settings/events.cljs

```diff
@@ -56,6 +57,10 @@
             {:profile.settings/webview-debug-changed value}
             (profile-update :webview-debug (boolean value) {})))
 
+(rf/reg-event-fx :profile.settings/change-test-networks-enabled
+ (fn [_ [value]]
+   {:fx [[:dispatch [:profile.settings/profile-update :test-networks-enabled? (boolean value) {}]]]}))
```

Comment:
I don't see much of a need for this new event if it's just re-dispatching and passing `value` converted to a boolean.

We can create a more useful event, like `:profile.settings/toggle-test-networks` and have it read from the app db itself. So the view wouldn't even need to pass `value`, since that's already in the global state.

Or just consume `:profile.settings/profile-update` directly in the view.

```diff
modified   src/status_im/ui/screens/advanced_settings/views.cljs
@@ -101,8 +101,7 @@
      :accessibility-label :test-networks-enabled
      :container-margin-bottom 8
      :on-press
-     #(re-frame/dispatch
-       [:profile.settings/change-test-networks-enabled (not test-networks-enabled?)])
+     #(re-frame/dispatch [:profile.settings/toggle-test-networks])
      :accessory :switch
      :active test-networks-enabled?}
     {:size :small
modified   src/status_im2/contexts/profile/settings/events.cljs
@@ -61,6 +61,11 @@
  (fn [_ [value]]
    {:fx [[:dispatch [:profile.settings/profile-update :test-networks-enabled? (boolean value) {}]]]}))
 
+(rf/reg-event-fx :profile.settings/toggle-test-networks
+ (fn [{:keys [db]}]
+   (let [value (get-in db [:profile/profile :test-networks-enabled?])]
+     {:fx [[:dispatch [:profile.settings/profile-update :test-networks-enabled? (not value) {}]]]})))
```